src/k8s/pkg/utils/k8s/node.go

```diff
@@ -0,0 +1,91 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+)
+
+// DrainNode drains the specified node by evicting its pods gracefully.
+func (c *Client) DrainNode(ctx context.Context, node string) error {
+	pods, err := c.CoreV1().Pods("").List(ctx, metav1.ListOptions{
+		FieldSelector: "spec.nodeName=" + node,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to get pods for node %s: %w", node, err)
+	}
+	for _, pod := range pods.Items {
+		if pod.Namespace == "kube-system" {
+			continue
+		} else {
+			if err := c.EvictPod(ctx, pod.Namespace, pod.Name); err != nil {
+				return fmt.Errorf("failed to evict pod %s from namespace %s: %w", pod.Name, pod.Namespace, err)
+			}
+		}
+	}
```

Comment:
this scares me a bit, is this our responsibility?

Reply:
let's not do this - as discussed offline.